qemu/qemu.go

```diff
@@ -447,6 +465,15 @@ func (o *Options) Cmdline() ([]string, error) {
 		args = append(args, "-initrd", o.Initramfs)
 	}
 
+	if o.OVMF != nil {
```

Comment:
This is nice! But not necessary:

```go
func WithOVMF(secbootPath, varsPath string) Fn {
  return func(alloc *IDAllocator, opts *Options) error {
    opts.AppendQEMU(
      fmt.Sprintf("-drive if=pflash,format=raw,unit=0,file=%s,readonly=on", secbootPath),
      fmt.Sprintf("-drive if=pflash,format=raw,unit=1,file=%s", varsPath),
    )
    return nil
  }
}
```

Having something to configure machine options and memory would probably be useful too, I haven't gotten around to those basics yet.

Reply:
Ah yes, then we could rather call those options. But we would need something to deduplicate the args line or is there a priority in the qemu flags?

Reply:
> Ah yes, then we could rather call those options. But we would need something to deduplicate the args line or is there a priority in the qemu flags?

Sorry, not sure I understand. Dedup would only be necessary if the user passed some conflicting args in VMTEST_QEMU or passed WithOVMF twice, right?

Reply:
> to add any arbitrary QEMU command-line args you want on a per-test basis.

I'm aware and what I'm currently doing, but considering you need more options for this I think it would be simpler to provide more abstract and easier to use knobs for this.

Reply:
I'll think on -m and -machine. I was mostly planning on letting the QEMU binary provide error messages for things like this, as there's probably tons of conflicting cmdline invocations possible that we can't all enumerate here. May be worth writing some tests and seeing what the handling of double -m or double -machine is in QEMU

Reply:
> > to add any arbitrary QEMU command-line args you want on a per-test basis.
> 
> I'm aware and what I'm currently doing, but considering you need more options for this I think it would be simpler to provide more abstract and easier to use knobs for this.

I agree, though I'm wondering what the best place for them is. Maybe a qemu/firmware package? I already started a qemu/network that needs quite a bit of improvement. I was hoping not all of these extensions would be in the main `qemu` package, as it makes the documentation overview a bit harder to read. (But I haven't been very serious about that separation yet, as you can tell...)